dashboard/database.py

```diff
@@ -131,25 +131,18 @@ def get_sales_by_tag(_conn: Connection, n: int = 5) -> pd.DataFrame:
     print("Counting sales by tag...")
 
     query = """
-        SELECT TG.name AS tag, COUNT(*) AS total_sales
-        FROM tag AS TG
-        JOIN album_tag_assignment AS ATG
-        ON ATG.tag_id =TG.tag_id
-        JOIN track_tag_assignment AS TGA
-        ON TGA.tag_id = TG.tag_id
-        JOIN album AS A
-        USING(album_id)
-        JOIN track AS T
-        USING(track_id)
-        JOIN album_purchase AS AP
-        ON AP.album_id = A.album_id
-        JOIN track_purchase AS TP
-        ON TP.track_id = T.track_id
-        GROUP BY tag
-        ORDER BY total_sales DESC
-        LIMIT %s
-        ;
-        """
+        SELECT t.name,
+    COUNT(DISTINCT ap.album_purchase_id) + COUNT(DISTINCT tp.track_purchase_id) AS total_sales
+    FROM tag AS t
+    LEFT JOIN album_tag_assignment AS ata ON t.tag_id = ata.tag_id
+    LEFT JOIN album_purchase AS ap ON ata.album_id = ap.album_id
+    LEFT JOIN track_tag_assignment AS tta ON t.tag_id = tta.tag_id
+    LEFT JOIN track_purchase AS tp ON tta.track_id = tp.track_id
+    GROUP BY t.name
+    ORDER BY total_sales DESC
+    LIMIT %s
+    ;
+    """
```

Comment:
Why was this query changed?

Reply:
The first query was returning incorrect data. Zhi spotted it and gave me the correct query. It uses `DISTINCT`, `LEFT JOIN` and `ON` instead

Reply:
I see, good change!